shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/AutoIntervalShardingAlgorithm.java

```diff
@@ -64,16 +65,16 @@
     public void init() {
         dateTimeLower = getDateTime(DATE_TIME_LOWER_KEY);
         shardingSeconds = getShardingSeconds();
-        autoTablesAmount = (int) (Math.ceil(parseDate(props.getProperty(DATE_TIME_UPPER_KEY)) / shardingSeconds) + 2);
+        autoTablesAmount = (int) (Math.ceil((double) (parseDate(props.getProperty(DATE_TIME_UPPER_KEY)) / shardingSeconds)) + 2);
```

Comment:
Question: `parseDate` return `long` (just as PR says), is `(double)` casting necessary?
And I found there's `(float)` casting in `doSharding` method, should it be `(double)` too?

Reply:
> Question: `parseDate` return `long` (just as PR says), is `(double)` casting necessary? And I found there's `(float)` casting in `doSharding` method, should it be `(double)` too?

For the change of the `Math.ceil()` parameter, because the input parameter of the `Math.ceil()` function is specified as `double`, so I do an explicit rather than an implicit division result of the `long` type The cast. This doesn't cause problems with the program, but I think explicit casts are good for IDE (this is reflected in `IntelliJ IDEA` as `Integer division in floating point context`) checks. I made synchronous changes , at the same time I did two tests, both with the same result.
````java
public void testFirst() {
DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        LocalDateTime dateTimeLower = LocalDateTime.parse("2020-01-01 00:00:00", DATE_TIME_FORMAT);
        System.out.println("dateTimeLower is----------------->" + dateTimeLower);
        int autoTablesAmount = (int) (Math.ceil((double) (Duration.between(dateTimeLower,
                LocalDateTime.parse("2022-12-28 00:00:00", DATE_TIME_FORMAT)).toMillis() / 1000 / shardingSeconds)) + 2);
        System.out.println("autoTablesAmount is---------------------------->" + autoTablesAmount);
        LocalDateTime dateValue = LocalDateTime.parse("2021-10-11 00:00:00", DATE_TIME_FORMAT);
        System.out.println("dateValue is-------------------->" + dateValue);
        long shardingValue = Duration.between(dateTimeLower, dateValue).toMillis() / 1000;
        System.out.println("shardingValue is------------------->" + shardingValue);
        String position = new DecimalFormat("0.00").format((double) shardingValue / shardingSeconds);
        System.out.println("position is-------------------------->" + position);
        int tableNameSuffix = Math.min(Math.max(0, (int) Math.ceil(Double.parseDouble(position))), autoTablesAmount - 1);
        System.out.println("tableNameSuffix is--------------->" + tableNameSuffix);
    }
````
Another test is the algorithm test before submitting the PR.
````java
public void testSecond() {
        long shardingSeconds = 604800L;
        DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        LocalDateTime dateTimeLower = LocalDateTime.parse("2020-01-01 00:00:00", DATE_TIME_FORMAT);
        System.out.println("dateTimeLower is----------------->" + dateTimeLower);
        int autoTablesAmount = (int) (Math.ceil(Duration.between(dateTimeLower,
                LocalDateTime.parse("2022-12-28 00:00:00", DATE_TIME_FORMAT)).toMillis() / 1000 / shardingSeconds) + 2);
        System.out.println("autoTablesAmount is---------------------------->" + autoTablesAmount);
        LocalDateTime dateValue = LocalDateTime.parse("2021-10-11 00:00:00", DATE_TIME_FORMAT);
        System.out.println("dateValue is-------------------->" + dateValue);
        long shardingValue = Duration.between(dateTimeLower, dateValue).toMillis() / 1000;
        System.out.println("shardingValue is------------------->" + shardingValue);
        String position = new DecimalFormat("0.00").format((float) shardingValue / shardingSeconds);
        System.out.println("position is-------------------------->" + position);
        int tableNameSuffix = Math.min(Math.max(0, (int) Math.ceil(Float.parseFloat(position))), autoTablesAmount - 1);
        System.out.println("tableNameSuffix is--------------->" + tableNameSuffix);
    }
````

Reply:
That's OK